LD52/Assets/Scripts/Player/Laser.cs

```diff
@@ -0,0 +1,124 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+[RequireComponent(typeof(LineRenderer), typeof(PolygonCollider2D))]
+
+public class Laser : MonoBehaviour
+{
+	private LineRenderer lr;
+    private List<Transform> nodes;
+    private List<Vector2> colliderPoints;
+    private PolygonCollider2D pc2d;
+
+    public Vector2 StartPosition { get { return nodes[0].position; } }  // world space
+    public Vector2 EndPosition {  // world space
+        get 
+        { 
+            return nodes[1].position; 
+        } 
+        set 
+        { 
+            nodes[1].position = value; 
+        } 
+    }
+    public Transform StartPositionTransform { get { return nodes[0]; } }
+    public Transform EndPositionTransform { get { return nodes[1]; } }
+
+    private Coroutine shootCoroutine;
+
+    void Awake()
+    {
+        GameObject endObject = new GameObject("LaserEnd");
```

Comment:
Is this a prefab?

Reply:
No, it's an empty `GameObject` that we use to denote the end of the laser.